test/foundry/Confiscate.t.sol

```diff
@@ -0,0 +1,290 @@
+        uint256 aliceAssets,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory justification,
+        uint256 certifyUntil,
+        uint256 referenceBlockNumber
+    ) external {
+        shareRatio = bound(shareRatio, 1, 1e18);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        referenceBlockNumber = bound(referenceBlockNumber, 1, block.number);
+        certifyUntil = bound(certifyUntil, 1, 1e6 - 1);
+
+        vm.assume(alice != bob);
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+        vault.grantRole(vault.CONFISCATOR(), alice);
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Call the certify function
+        vault.certify(certifyUntil, referenceBlockNumber, false, justification);
+
+        //set upperBound for assets so it does not overflow while calculating fixedPointDiv or fixedPointMul
+        uint256 upperBound = type(uint256).max / 1e18;
+        // Assume that aliceAssets is less than totalSupply
+        aliceAssets = bound(aliceAssets, 1, upperBound);
+
+        vault.deposit(aliceAssets, bob, shareRatio, justification);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(bob, alice, aliceAssets);
+
+        vault.confiscateShares(bob, justification);
+        vm.stopPrank();
+    }
+
+    /// Test to checks ConfiscateReceipt is NOT emitted on zero balance
+    function testConfiscateReceiptOnZeroBalance(
+        uint256 fuzzedKeyAlice,
+        uint256 fuzzedKeyBob,
+        string memory assetName,
+        string memory assetSymbol,
+        bytes memory justification
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        vm.assume(alice != bob);
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.CONFISCATOR(), alice);
+
+        // Stop recording logs
+        Vm.Log[] memory logs = vm.getRecordedLogs();
+        vault.confiscateReceipt(bob, 1, justification);
```

Comment:
why 1?

Reply:
I do one deposit and have one receipt with id =1, If i fuzz id here, and the id is different than 1, there will be zero balance and event will not be emitted.